src/til/ut_til/BitmapTests.cpp

```diff
@@ -0,0 +1,329 @@
+        {
+            auto fn = [&]() {
+                map.set(til::point{ 10, 10 });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+
+        Log::Comment(L"2.) SetRectangle out of bounds.");
+        {
+            auto fn = [&]() {
+                map.set(til::rectangle{ til::point{
+                                            2,
+                                            2,
+                                        },
+                                        til::size{ 10, 10 } });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+
+        Log::Comment(L"3.) ResetPoint out of bounds.");
+        {
+            auto fn = [&]() {
+                map.reset(til::point{ 10, 10 });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+
+        Log::Comment(L"4.) ResetRectangle out of bounds.");
+        {
+            auto fn = [&]() {
+                map.reset(til::rectangle{ til::point{
+                                              2,
+                                              2,
+                                          },
+                                          til::size{ 10, 10 } });
+            };
+
+            VERIFY_THROWS_SPECIFIC(fn(), wil::ResultException, [](wil::ResultException& e) { return e.GetErrorCode() == E_INVALIDARG; });
+        }
+    }
+
+    TEST_METHOD(Resize)
+    {
+        Log::Comment(L"Set up a bitmap with every location flagged.");
+        const til::size originalSize{ 2, 2 };
+        til::bitmap bitmap{ originalSize };
+
+        bitmap.set_all();
+
+        Log::Comment(L"Attempt resize to the same size.");
+        VERIFY_IS_FALSE(bitmap.resize(originalSize));
+
+        Log::Comment(L"Attempt resize to a new size.");
+        VERIFY_IS_TRUE(bitmap.resize(til::size{ 3, 3 }));
+    }
+
+    TEST_METHOD(Empty)
+    {
+        Log::Comment(L"When created, it should be empty.");
+        til::bitmap bitmap{ til::size{ 2, 2 } };
+        VERIFY_IS_TRUE(bitmap.empty());
+
+        Log::Comment(L"When it is modified with a set, it should be non-empty.");
+        bitmap.set(til::point{ 0, 0 });
+        VERIFY_IS_FALSE(bitmap.empty());
+
+        // We don't track it becoming empty again by resetting points because
+        // we don't know for sure it's entirely empty unless we seek through the whole thing.
+        // It's just supposed to be a short-circuit optimization for between frames
+        // when the whole thing has been processed and cleared.
+        Log::Comment(L"Resetting the same point, it will still report non-empty.");
+        bitmap.reset(til::point{ 0, 0 });
+        VERIFY_IS_FALSE(bitmap.empty());

```

Comment:
This makes empty misleading. :(

Reply:
I have an idea to fix this after discussing with Dustin last night.